chains/evm/contracts/ccvs/RampProxy.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+/// @notice RampProxy enables upgrades to CCVRamps without breaking existing references in token pools, receivers, and apps.
+/// The address of this contract will be referenced in the following places:
+///   - Users / apps will specify required and optional CCVs as part of ccipSend extraArgs.
+///   - Token pools will specify required and optional CCVs on both source and destination.
+///   - Receiver contracts will specify required and optional CCVs on destination.
+///   - CCVProxy will specify default and mandated CCVs for each destination.
+///   - CCVAggregator will specify default and mandated CCVs for each source.
+/// Each of these references should be to a RampProxy contract, not a CCVRamp directly.
+/// @dev On source, the CCVProxy will forward requests (i.e. getFee, forwardToVerifier) through this contract to the required CCVRamp.
+/// The same applies on destination. The CCVAggregator will forward requests (i.e. verifyMessage) through this contract to the required CCVRamp.
+/// To support this proxy, all future CCVRamp interfaces must have originalCaller defined as the first arg to each method.
+contract RampProxy {
+  error ZeroAddressNotAllowed();
+
+  event RampUpdated(address indexed oldRamp, address indexed newRamp);
+
+  /// @notice The address of the ramp contract.
+  address public s_ramp;
+
+  constructor(
+    address rampAddress
+  ) {
+    _setRamp(rampAddress);
+  }
+
+  function _setRamp(
+    address rampAddress
+  ) internal {
+    if (rampAddress == address(0)) {
+      revert ZeroAddressNotAllowed();
+    }
+    address oldRamp = s_ramp;
+    s_ramp = rampAddress;
+    emit RampUpdated(oldRamp, rampAddress);
+  }
+
+  // The fallback function forwards all calls to the ramp contract via delegatecall.
```

Comment:
This is missing the very critical information that it overrides the first argument with the msg.sender.

Reply:
https://github.com/smartcontractkit/chainlink-ccip/pull/1182/commits/8f1762ef84e1348c20ab970fbc61f97ba2327bf7